File: GoodCodes/GoodCodes_mint_old_backup/incomplete_type_of_array.py
def findMax(h,l,iarr):
	mx = ((int)((h+l)/2))+1
	while( l < h ):
		m = (int)((h+l)/2)
		if( m+1 < len(iarr) ):
			#print("yes111")

			if( iarr[m] > iarr[mx] ):
				# print("yes222")
				if( iarr[m+1] < iarr[m] ):
					h = m
					mx = m
				else:#iarr[m+1] > m
					l = m+1
					mx = m+1

			elif( iarr[m] < iarr[mx] ):
				if( iarr[m+1] < iarr[m] ):
					h = m-1
				else:#iarr[m+1] > m
					l = m+1
					if( iarr[m+1] > iarr[mx] ):
						mx = m+1

			elif( iarr[m] == iarr[mx] ):
				if( iarr[m+1] < iarr[m] ):
					h = m
				else:#iarr[m+1] > m
					l = m+1
					mx = m+1
				break

			else:
				break
		else:
			#print("just something : ",iarr[m])
			break
	return mx

t = (int)(input())
while(t > 0):
	input()
	iarr=list(map(int,input().split()))
	#for(mx = -1, h = iarr.length-1, l = 0, m = (h+l)/2 l<highmid = (h+l)/2):
	mx = findMax(len(iarr)-1, 0, iarr)

	if( mx > 0 and mx < len(iarr)-1 ):
		while( iarr[mx+1] > iarr[mx] ):
			mx = mx+1
		while( iarr[mx-1] > iarr[mx] ):
			mx = mx-1
	elif( mx == 0 ):
		if iarr[len(iarr)-1] > iarr[mx]:
			mx = len(iarr)-1
			while( iarr[mx-1] > iarr[mx] ):
				mx = mx-1
		# temp = findMax( len(iarr)-1, (int)(len(iarr)/2), iarr)
		# print(temp)
		# mx = (mx,temp)[ iarr[temp] > iarr[mx] ]
	elif( mx == len(iarr)-1 ):
		if iarr[0] > iarr[mx]:
			mx = 0
			while( iarr[mx+1] > iarr[mx] ):
				mx = mx+1
		# temp = findMax( (int)(len(iarr)/2), 0, iarr)
		# mx = (mx,temp)[ iarr[temp] > iarr[mx] ]
	# print(mx)
	# print(len(iarr)-1)
	# type1 = ( (3, 2)[mx==0], (4, 1)[len(iarr)-1==mx] )[ iarr[0] < iarr[1] and iarr[1] < iarr[2]]


	if (iarr[0]<iarr[1]) == (iarr[len(iarr)-2]<iarr[len(iarr)-1]):
		type1 = (2, 1)[iarr[0] < iarr[1]]
	else:
		if mx==0:
			type1 = (2, 1)[iarr[len(iarr)-2] < iarr[len(iarr)-1]]
			
		else:
			type1 = (2, 1)[iarr[0] < iarr[1]]
	type2 = (type1, 5-type1)[mx not in (0, len(iarr)-1)]
	print(iarr[mx],type2)
	t=t-1;

File: GoodCodes/GoodCodes/problematic/incomplete_type_of_array.java
// 10
// 5
// 1 2 3 4 5
// 6
// 1 2 3 4 5 6
// 5
// 5 4 3 2 1
// 6
// 6 5 4 3 2 1
// 5
// 4 5 1 2 3
// 6
// 4 5 6 1 2 3
// 5
// 3 2 1 5 4
// 6
// 3 2 1 6 5 4
// 5
// 4 5 1 2 3
// 6
// 3 2 1 6 5 4



// if found increment then ascendubgg
// else descemminf

// if max element1 is at extreme than normal else rotated

// type = (2 ,1)[ iarr[0] < iarr[1] ]
// if max!=iarr[iarr.length-1]
// 	type = 5-type

import java.io.*;
import java.lang.*;
import java.util.*;
import java.util.stream.*;
public class GFG{
	public static void main(String[] args)throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int[] iarr;
		Set<Integer> ihash;
		boolean flag;
		int max, high, low, mid;
		for(int i, j, k, t = GFG.readX(br)[0]; t>0; t--)
		{
			br.readLine();
			iarr = GFG.readX(br);
			for(max = -1, high = iarr.length-1, low = 0, mid = (high+low)/2; low<high;mid = (high+low)/2)
			{
				if(mid+1 < iarr.length)
				{
					//System.out.println("yes111");
					if(iarr[mid] > max)
					{
						// System.out.println("yes222");
						if(iarr[mid+1] < iarr[mid])
						{
							high = mid-1;
							max = iarr[mid];
						}
						else//iarr[mid+1] > mid
						{
							low = mid+1;
							max = iarr[mid+1];
						}
					}
					else if(iarr[mid] < max)
					{
						if(iarr[mid+1] < iarr[mid])
						{
							low = mid-1;
						}
						else//iarr[mid+1] > mid
						{
							low = mid+1;
						}
					}
					else if(iarr[mid] == max)
					{
						if(iarr[mid+1] < iarr[mid])
						{
							high = mid;
							break;
						}
						else//iarr[mid+1] > mid
						{
							low = mid+1;
							max = iarr[mid+1];
							break;
						}
					}
					else break;
				}
				else
				{
					System.out.println("just something : "+iarr[mid]);
					break;
				}
			}
			System.out.println(max);
		}
	}
	public static int[] readX(BufferedReader br)throws IOException{
		return Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
	}
	// public static Set<Integer> readX2(BufferedReader br)throws IOException{
	// 	return Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).boxed().collect(Collectors.toCollection(HashSet::new));
	// }
}